api/capi/src/nntrainer.cpp

```diff
@@ -258,6 +265,27 @@ int ml_nnmodel_add_layer(ml_nnmodel_h model, ml_nnlayer_h layer) {
   return status;
 }
 
+int ml_nnmodel_set_optimizer(ml_nnmodel_h model, ml_nnopt_h optimizer) {
+  int status = ML_ERROR_NONE;
+  ml_nnmodel *nnmodel;
+  ml_nnopt *nnopt;
+
+  ML_NNTRAINER_CHECK_MODEL_VALIDATION(nnmodel, model);
+  ML_NNTRAINER_CHECK_OPT_VALIDATION(nnopt, optimizer);
+
+  std::shared_ptr<nntrainer::NeuralNetwork> NN;
+  std::shared_ptr<nntrainer::Optimizer> opt;
+
+  NN = nnmodel->network;
+  opt = nnopt->optimizer;
+
+  returnable f = [&]() { return NN->setOptimizer(opt); };
+
+  status = nntrainer_exception_boundary(f);
+
```

Comment:
Suggestion: if fail setting optimizer, destruct optimizer?

Reply:
IMO, this might be overkill.

Reply:
Yeah you're right. It's not creating optimizer so it shouldn't delete

Reply:
Yes, if this is to be added, this would extend to deletion of optimizer if set property fails and spiral from there.